Problem: pulp_rpm's need for system-wide packages conflicts

mikedep333 opened the pull request:
Problem: pulp_rpm's need for system-wide packages conflicts with having a newer version of pip

Solution: Eliminate using system-wide packages.

This partially reverts commit 14c64863c87432edd507e1f396f644231a207340.
"Problem: pulp_use_system_wide_pkgs is often overlooked"

Fixes: #6267

mikedep333 committed:
Problem: pulp_rpm's need for system-wide packages conflicts with having a newer version of pip

Solution: Eliminate using system-wide packages.

This partially reverts commit 14c64863c87432edd507e1f396f644231a207340.
"Problem: pulp_use_system_wide_pkgs is often overlooked"

fixes: #6267

mikedep333 commented:
This was my previous attempt at eliminating the need for system-wide packages:
https://github.com/pulp/pulp_rpm_prerequisites/pull/16

I recall only being able to minimize them.

fao89 commented on .ansible-pulp_tox.sh:
```diff
@@ -9,6 +9,8 @@ if [ ! -e roles/pulp.pulp_rpm_prerequisites ]; then
   ln -s $GITHUB_WORKSPACE roles/pulp.pulp_rpm_prerequisites
 fi
 
+sed -i 's/pulp_use_system_wide_pkgs: false/pulp_use_system_wide_pkgs: true/g' roles/pulp/defaults/main.yml
```
it sets pulp_use_system_wide_pkgs to True

mikedep333 committed:
Problem: pulp_rpm_prerequisites is slow to install packages

Solution: Use yum module for RH family, the only supported family
currently, and fallback to package for others.

Also remove epel-release: We now rely on ansible-pulp to install
it beforehand due to how this role is now included, and since
packages are no longer being installed 1 by 1, it will not help.

[noissue]

fao89 reviewed (approved):
Great job! :tada:

mikedep333 commented:
@fabricio-aguiar I'm still trying to figure out what we should do for users that have an old install with system-wide.

I suggest we do it in a separate commit. I am running pulplift now. We may not need any changes at this time for centos7 due to the pip dependencies being the same, or roughly the same.

To determine what to do, I suggest we add a new upgrade molecule CI scenario: pulpcore 3.1.0 & pulp_rpm 3.1.0 -> latest of each.

The current upgrade scenario tests pulpcore 3.0.0 & pulp_file 0.1.0 -> pulpcore latest, pulp_file latest, and pulp_rpm latest (pulp_rpm is fresh installed.)

fao89 commented:
> @fabricio-aguiar I'm still trying to figure out what we should do for users that have an old install with system-wide.
> 
> I suggest we do it in a separate commit. I am running pulplift now. We may not need any changes at this time for centos7 due to the pip dependencies being the same, or roughly the same.
> 
> To determine what to do, I suggest we add a new upgrade molecule CI scenario: pulpcore 3.1.0 & pulp_rpm 3.1.0 -> latest of each.
> 
> The current upgrade scenario tests pulpcore 3.0.0 & pulp_file 0.1.0 -> pulpcore latest, pulp_file latest, and pulp_rpm latest (pulp_rpm is fresh installed.)

I think those PRs that I changed pyproject.toml would help old users with system-wide packages.
And I agree with your suggested scenario for pulp_rpm

mikedep333 commented:
@fabricio-aguiar Great! Please continue with them.
